pymodbus/transport/transport.py

```diff
@@ -231,6 +233,7 @@ def init_setup_connect_listen(self, host: str, port: int) -> None:
                 self.handle_new_connection,
                 host,
                 port,
+                sock=self.comm_params.socket,
```

Comment:
This affects both the server (listen) and client (connect), but you have not changed the server part.

This really needs to be tested, what happens if you pass none.